src/realm/object-store/shared_realm.cpp

```diff
@@ -910,6 +910,35 @@ void Realm::close()
     m_coordinator = nullptr;
 }
 
+bool Realm::delete_files(const std::string& realm_file_path)
+{
+    const auto& core_files =
+        DB::get_core_files(realm_file_path, DB::CoreFileType::StateFiles | DB::CoreFileType::TemporaryFiles);
+    auto lock_successful = DB::call_with_lock(realm_file_path, [&](auto) {
+        for (const auto& file : core_files) {
+            const auto& file_path = file.first;
+            const auto& is_folder = file.second;
+            if (is_folder) {
+                // The return value can be ignored since non-existing folders do not matter here.
+                // We do need to use the `try_remove_dir_recursive` instead of `remove_dir_recursive` to
+                // not throw in that case.
+                util::try_remove_dir_recursive(file_path); // Throws
+            }
+            else {
+                // The return value can be ignored since non-existing files do not matter here.
+                // We do need to use the `try_remove_dir_recursive` instead of `remove_dir_recursive` to
```

Comment:
this comment looks like copy/paste and doesn't seem to apply to the code below

Reply:
I did in fact copy/paste it and then not adjust the slight differences. e02c81924 addresses this better now with less text.